### PR TITLE
Vmij 300 test counterdown

### PR DESCRIPTION
**Pull request by millevlada**
see [#300]

### CODE REVIEW

File: web/src/main/frontend/src/components/TimeRemaining.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      this.props.onExpire();
+    }
+  }
+
+  formatRemainingTime(secondsRemaining) {
+    let min = Math.floor(secondsRemaining / 60);
+    let sec = secondsRemaining % 60;
+
+    if (min > 2) {
+      return `${min}min`;
```
Comment (iscekic) [inline web/src/main/frontend/src/components/TimeRemaining.js]: There should be a space between `}` and `min`

File: web/src/main/frontend/src/components/TimeRemaining.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    if (sec < 10) {
+      sec = '0' + sec;
+    }
+
+    if (min > 0) {
+      return `${min}min ${sec}sec`;
```
Comment (iscekic) [inline web/src/main/frontend/src/components/TimeRemaining.js]: Space after `}`

File: web/src/main/frontend/src/pages/TakeTest.js [hunk 3]
```diff
@@ -3,15 +3,30 @@ import {observer, inject} from 'mobx-react';
...
+
   componentDidMount() {
-    this.props.newTest.fetchGeneratedTest(this.props.match.params.testId);
+    this.props.newTest.initialize(
+      () => this.props.counterDown.secondsRemaining
+    );
+    this.props.newTest
+      .fetchGeneratedTest(this.props.match.params.testId)
+      .then(() => {
+        let remainingTime = this.props.newTest.test.remainingTime;
```
Comment (iscekic) [inline web/src/main/frontend/src/pages/TakeTest.js]: Can use destructuring here.

File: web/src/main/frontend/src/scss/components/_takeTest.scss [hunk 4]
```diff
@@ -102,3 +102,75 @@
...
   }
 }
+
+.take-test-counter__warning {
+  color: #ff0000; // red
+}
+
+.take-test-counter__alarm {
+  animation: 1s blinker linear infinite;
+  -webkit-animation: 1s blinker linear infinite;
```
Comment (iscekic) [inline web/src/main/frontend/src/scss/components/_takeTest.scss]: No need for prefixed versions

File: web/src/main/frontend/src/scss/components/_reviewTest.scss [hunk 5]
```diff
@@ -55,13 +55,14 @@
...
   min-height: fit-content;
-  padding: 0px 20% 0px 20%;
+  padding: 0px 2% 0px 2%;
   margin-bottom: 20px;
 }
 
 .review__container__counter {
   background-color: #f2f2f2;
   height: 100%;
+  padding: 0px 20%;
```
Comment (iscekic) [inline web/src/main/frontend/src/scss/components/_reviewTest.scss]: Please make sure these % paddings are ok for all window sizes

File: web/src/main/frontend/src/scss/components/_takeTest.scss [hunk 6]
```diff
@@ -102,3 +102,75 @@
...
+  -moz-animation: 1s blinker linear infinite;
+}
+
+@-moz-keyframes blinker {
```
Comment (iscekic) [inline web/src/main/frontend/src/scss/components/_takeTest.scss]: No need for prefixed versions

File: web/src/main/frontend/src/components/TimeRemaining.js (hunk 1 above)
Comment (millevlada) [reply web/src/main/frontend/src/components/TimeRemaining.js]: hm, idea was to have it stiched actually, like:
3min
or
2min 23sec

It was like that on other test sites ;)

so, I'll keep it ;)

File: web/src/main/frontend/src/scss/components/_takeTest.scss (hunk 4 above)
Comment (millevlada) [reply web/src/main/frontend/src/scss/components/_takeTest.scss]: Ok, I pasted that from some web article, indeed it works without these extras ;) Tnx.

File: web/src/main/frontend/src/scss/components/_reviewTest.scss (hunk 5 above)
Comment (millevlada) [reply web/src/main/frontend/src/scss/components/_reviewTest.scss]: should be fine

**Pull request merged**